components/org.wso2.bfsi.identity.extensions/src/test/java/org/wso2/bfsi/identity/extensions/util/IdentityCommonUtilsTest.java

```diff
@@ -0,0 +1,94 @@
+
+import java.security.cert.CertificateException;
+import java.security.cert.X509Certificate;
+import java.util.Arrays;
+
+/**
+ * Test class for IdentityCommonUtils.
+ */
+public class IdentityCommonUtilsTest {
+
+    @Test
+    public void testGetConsentIdFromScopes() {
+        String[] scopes = {"dummy-scope1", "dummy-scope2", "consent_id_ConsentId", "dummy-scope3", "BFSI_ConsentId",
+                "TIME_ConsentId", "x5t#_ConsentId"};
+        Assert.assertTrue(Arrays.toString(scopes).contains("consent_id"));
+        String[] modifiedScopes = IdentityCommonUtils.removeInternalScopes(scopes);
+        Assert.assertFalse(Arrays.toString(modifiedScopes).contains("consent_id"));
+        Assert.assertFalse(Arrays.toString(modifiedScopes).contains("BFSI_"));
+        Assert.assertFalse(Arrays.toString(modifiedScopes).contains("TIME_"));
+        Assert.assertFalse(Arrays.toString(modifiedScopes).contains("x5t#_"));
+    }
+
+    @Test(description = "when valid transport cert, return x509 certificate")
+    public void testParseCertificate() throws ConsentManagementException {
+        Assert.assertNotNull(IdentityCommonUtils.parseCertificate(TestConstants.TEST_CLIENT_CERT));
+    }
+
+    @Test (expectedExceptions = ConsentManagementException.class)
+    public void testParseCertificateWithInvalidCert() throws ConsentManagementException {
+        Assert.assertNull(IdentityCommonUtils.parseCertificate("-----INVALID CERTIFICATE-----"));
+    }
+
+    @Test
+    public void testParseCertificateWithInvalidBase64CharactersCert() throws ConsentManagementException {
+        Assert.assertNotNull(IdentityCommonUtils.parseCertificate(TestConstants.WRONGLY_FORMATTED_CERT));
+    }
+
+    @Test
+    public void testParseCertificateWithEmptyCert() throws ConsentManagementException {
+        Assert.assertNull(IdentityCommonUtils.parseCertificate(""));
+    }
+
+    @Test(description = "when certificate expired, return true")
+    public void testIsExpiredWithExpiredCert() throws ConsentManagementException {
+        X509Certificate testCert = IdentityCommonUtils.parseCertificate(TestConstants.EXPIRED_SELF_CERT);
+        Assert.assertNotNull(testCert);
+        Assert.assertTrue(hasExpired(testCert));
+    }
+
+    @Test(description = "when valid certificate, return false")
+    public void testIsExpired() throws ConsentManagementException {
+        X509Certificate testCert = IdentityCommonUtils.parseCertificate(TestConstants.TEST_CLIENT_CERT);
+        Assert.assertNotNull(testCert);
+        Assert.assertFalse(hasExpired(testCert));
+    }
+
+    /**
+     * Test util method to check cert expiry.
+     *
+     * @param peerCertificate
+     * @return
+     */
+    public static boolean hasExpired(X509Certificate peerCertificate) {
+        try {
+            peerCertificate.checkValidity();
+        } catch (CertificateException e) {
+            return true;
+        }
+        return false;
+    }
+}
```

Comment:
Add new line, check other places as well.

Reply:
Line is already there, hence resolving comments